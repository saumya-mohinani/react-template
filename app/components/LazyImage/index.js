import styled from 'styled-components';
import PropTypes from 'prop-types';
import React from 'react';

const ImageArt = styled.img`
  && {
    object-fit: contain;
    height: 250px;
  }
`;

const useProgressiveImg = (lowQualitySrc, highQualitySrc) => {
  const [src, setSrc] = React.useState(lowQualitySrc);
  React.useEffect(() => {
    setSrc(lowQualitySrc);
    const img = new Image();
    img.src = highQualitySrc;
    img.onload = () => {
      setSrc(highQualitySrc);
    };
  }, [lowQualitySrc, highQualitySrc]);
  return [src, { blur: src === lowQualitySrc }];
};

const LazyImage = ({ source }) => {
  const [src] = useProgressiveImg(`https://placeholder.pics/svg/250/DEDEDE/555555/loading...`, source);
  return <ImageArt src={src} />;
};

LazyImage.propTypes = {
  source: PropTypes.string.isRequired
};

export default LazyImage;
