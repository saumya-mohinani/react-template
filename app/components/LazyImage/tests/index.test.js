/**
 *
 * Tests for LazyImage
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import LazyImage from '../index';
import logo from '@images/logo.png';

describe('<LazyImage />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<LazyImage lowResUrl={logo} highResUrl={logo} />);
    expect(baseElement).toMatchSnapshot();
  });
});
