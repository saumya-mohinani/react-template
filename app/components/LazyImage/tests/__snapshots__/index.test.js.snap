// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyImage /> should render and match the snapshot 1`] = `
<body>
  <div>
    <img
      class="LazyImage__ImageArt-sc-1wx4egn-0 jfApRO"
      src="IMAGE_MOCK"
    />
  </div>
</body>
`;
